src/object.rs

```diff
@@ -673,33 +675,32 @@ impl<T: IsA<Object> + SetValue> ObjectExt for T {
         let property_name = property_name.into();
         let property_value = value.to_value();
 
-        unsafe {
-            let object = self.to_glib_none().0;
-            let klass = (*object).g_type_instance.g_class as *mut gobject_ffi::GObjectClass;
-            let pspec = gobject_ffi::g_object_class_find_property(klass, property_name.to_glib_none().0);
-            if pspec.is_null() {
+        let pspec = match self.find_property(property_name) {
+            Some(pspec) => pspec,
+            None => {
                 return Err(BoolError("property not found"));
-            }
+            },
+        };
 
-            let writeable = (*pspec).flags & gobject_ffi::G_PARAM_WRITABLE != 0;
-            if !writeable {
-                return Err(BoolError("property is not writeable"));
-            }
+        if !pspec.get_flags().contains(::ParamFlags::WRITABLE) {
```

Comment:
Maybe flags need check for `CONSTRUCT_xxx`, some properties have WRITABLE and CONSTRUCT_ONLY, so actually not writable

Reply:
Thanks, good idea :) Will change in a few minutes

Reply:
done